client/app/components/queries/api-key-dialog.js

```diff
@@ -4,26 +4,46 @@ const ApiKeyDialog = {
 </div>
 <div class="modal-body">
     <h5>API Key</h5>
-    <pre>{{$ctrl.apiKey}}</pre>
+    <div class="form-group">
+        <pre>{{$ctrl.api_key}}</pre>
+        <div ng-if="$ctrl.canEdit">
+            <button class="btn btn-default" ng-click="$ctrl.regenerateQueryApiKey()" ng-disabled="$ctrl.disableRegenerateApiKeyButton">Regenerate</button>
+        </div>
+    </div>
 
     <h5>Example API Calls:</h5>
 
     <div>
         Results in CSV format:
 
-        <pre>{{$ctrl.csvUrl}}</pre>
+        <pre>{{$ctrl.csvUrlBase + $ctrl.api_key}}</pre>
 
         Results in JSON format:
 
-        <pre>{{$ctrl.jsonUrl}}</pre>
+        <pre>{{$ctrl.jsonUrlBase + $ctrl.api_key}}</pre>
     </div>
 </div>`,
-  controller(clientConfig) {
+  controller($http, clientConfig, currentUser) {
     'ngInject';
 
-    this.apiKey = this.resolve.query.api_key;
-    this.csvUrl = `${clientConfig.basePath}api/queries/${this.resolve.query.id}/results.csv?api_key=${this.apiKey}`;
-    this.jsonUrl = `${clientConfig.basePath}api/queries/${this.resolve.query.id}/results.json?api_key=${this.apiKey}`;
+    this.canEdit = currentUser.id === this.resolve.query.user.id || currentUser.hasPermission('admin');
+    this.disableRegenerateApiKeyButton = false;
+    this.api_key = this.resolve.query.api_key;
+    this.csvUrlBase = `${clientConfig.basePath}api/queries/${this.resolve.query.id}/results.csv?api_key=`;
+    this.jsonUrlBase = `${clientConfig.basePath}api/queries/${this.resolve.query.id}/results.json?api_key=`;
+
+    this.regenerateQueryApiKey = () => {
+      this.disableRegenerateApiKeyButton = true;
+      $http
+        .post(`api/queries/${this.resolve.query.id}/regenerate_api_key`)
+        .success((data) => {
+          this.api_key = data.api_key;
```

Comment:
1. Regenerate API Key
1. Close dialog
1. Reopen dialog (API Key isn't updated)

Need to update the query object, but I have no idea.

Reply:
You should update the `query.api_key` for it to persist.

Reply:
Thanks. Fixed it.